src/pages/Loans/LoansOverview/hooks/use-get-account-health-factor.tsx

```diff
@@ -0,0 +1,86 @@
+import { CurrencyExt } from '@interlay/interbtc-api';
+import { MonetaryAmount } from '@interlay/monetary-js';
+import Big from 'big.js';
+import { useCallback } from 'react';
+
+import { LoanAction } from '@/types/loans';
+import { useGetAccountLoansOverview } from '@/utils/hooks/api/loans/use-get-account-loans-overview';
+import { useGetLoanAssets } from '@/utils/hooks/api/loans/use-get-loan-assets';
+import { useGetPrices } from '@/utils/hooks/api/use-get-prices';
+
+import { calcutateCollateralBorrowedAmountUSD } from '../utils/math';
+
+const calculateHealthFactor = (borrowAmountUSD: Big, collateralAmountUSD: Big) =>
+  borrowAmountUSD.gt(0) ? collateralAmountUSD.div(borrowAmountUSD).toNumber() : Infinity;
+
+const getCurrencyHealthFactor = (borrowAmountUSD?: Big, collateralAmountUSD?: Big): number | undefined => {
+  if (borrowAmountUSD === undefined || collateralAmountUSD === undefined) {
+    return undefined;
+  }
+
+  return calculateHealthFactor(borrowAmountUSD, collateralAmountUSD);
+};
+
+type LoanActionData = { type: LoanAction; amount: MonetaryAmount<CurrencyExt> };
+
+interface UseLoansHealthFactor {
+  healthFactor: number | undefined;
+  getHealthFactor: (loanAction: LoanActionData) => number | undefined;
+}
+
+const useLoansHealthFactor = (): UseLoansHealthFactor => {
```

Comment:
Should be renamed to `useGetAccountHealthFactor` to match the filename and other hook.